File: C++/LeetCode/Array/2011_Final_Value_of_Variable_After_Performing_Operations.cpp
class Solution {
public:
    int finalValueAfterOperations(vector<string>& operations) {
        
        int value = 0;
        
        for(string operation : operations)
        {
            for(char character : operation)
            {
                if(character == '+') 
                {
                    value += 1;
                    break;
                }
                else if(character == '-')
                {
                    value -= 1;
                    break;
                }
            }
        }
        
        return value;
    }
};

File: Java/LeetCode/2011_Final_Value_of_Variable_After_Performing_Operations.java
class Solution {
    private int processOperation(String operation) {
        var firstIndex = 0;
        var operationLastIndex = operation.length() - 1;

        if (isDecrement(operation, firstIndex, operationLastIndex)) {
            return -1;
        }

        if (isIncrement(operation, firstIndex, operationLastIndex)) {
            return 1;
        }

        return 0;
    }

    private boolean isIncrement(String operation, int firstIndex, int lastIndex) {
        return operation.charAt(firstIndex) == '+' 
            || operation.charAt(lastIndex) == '+';
    }

    private boolean isDecrement(String operation, int firstIndex, int lastIndex) {
        return operation.charAt(firstIndex) == '-' 
            || operation.charAt(lastIndex) == '-';
    }

    public int finalValueAfterOperations(String[] operations) {
        var operationLength = operations.length;
        var finalCount = 0;
        
        for (var operation: operations) {
            finalCount += processOperation(operation);
        }

        return finalCount;
    }
}